internal/setup/setup.go

```diff
@@ -623,3 +650,75 @@ func Select(release *Release, slices []SliceKey) (*Selection, error) {
 
 	return selection, nil
 }
+
+// convertPathInfoToYamlPath converts a PathInfo object to a yamlPath object.
+func convertPathInfoToYamlPath(pi *PathInfo) (*yamlPath, error) {
+	if pi == nil {
+		return nil, nil
+	}
+	path := &yamlPath{
+		Mode:    yamlMode(pi.Mode),
+		Mutable: pi.Mutable,
+		Until:   pi.Until,
+		Arch:    yamlArch{List: pi.Arch},
+	}
+	switch pi.Kind {
+	case DirPath:
+		path.Dir = true
+	case CopyPath:
+		path.Copy = pi.Info
+	case TextPath:
+		path.Text = &pi.Info
+	case SymlinkPath:
+		path.Symlink = pi.Info
+	case GlobPath:
+	default:
+		return nil, fmt.Errorf("unrecognised PathInfo type: %s", pi.Kind)
+	}
+	return path, nil
+}
+
+// convertSliceToYamlSlice converts a Slice object to a yamlSlice object.
+func convertSliceToYamlSlice(s *Slice) (*yamlSlice, error) {
+	if s == nil {
+		return nil, nil
+	}
+	slice := &yamlSlice{
+		Essential: make([]string, len(s.Essential)),
+		Contents:  make(map[string]*yamlPath, len(s.Contents)),
+		Mutate:    s.Scripts.Mutate,
+	}
+	for idx, sliceKey := range s.Essential {
+		slice.Essential[idx] = sliceKey.String()
```

Comment:
`i` is more typical for Go loops. Also no need for `sliceKey` as everything's about slices here; just `key` should be fine:

```suggestion
	for i, key := range s.Essential {
		slice.Essential[i] = key.String()
```

Reply:
Thanks.